src/bid/encoding.rs

```diff
@@ -15,14 +17,15 @@ use poseidon252::{sponge::sponge::*, StorageScalar};
 // Type 4 will be u32
 // Type 5 will be PoseidonCipher
 // Byte-types are treated in Little Endian.
-// The purpose of this set of flags is to avoid collision between different structures
+// The purpose of this set of flags is to avoid collision between different
+// structures
 const TYPE_FIELDS: [u8; 32] = *b"53313110000000000000000000000000";
 
 /// Encodes a `StorageBid` in a `StorageScalar` form by applying the correct
 /// encoding methods and collapsing it into a `StorageScalar` which can be then
 /// stored inside of a `kelvin` tree data structure.
-impl Into<StorageScalar> for Bid {
-    fn into(self) -> StorageScalar {
+impl From<&Bid> for StorageScalar {
+    fn from(bid: &Bid) -> StorageScalar {
```

Comment:
Just FYI you can use `Self` instead here, so `fn from(bid: &Bid) -> Self`.

Reply:
I know :)